go-controller/observability-lib/sampledecoder/sample_decoder.go

```
@@ -133,90 +135,88 @@ func (d *SampleDecoder) DecodeCookieIDs(obsDomainID, obsPointID uint32) (string,
-		case libovsdbops.BaselineAdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("baseline admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastNamespaceOwnerType:
-			msg = fmt.Sprintf("multicast in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastClusterOwnerType:
-			msg = fmt.Sprintf("cluster multicast policy, direction %s", o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNodeOwnerType:
-			msg = "default allow from local node policy, direction ingress"
-		case libovsdbops.NetworkPolicyOwnerType:
-			msg = fmt.Sprintf("network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNamespaceOwnerType:
-			msg = fmt.Sprintf("network policies isolation in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.EgressFirewallOwnerType:
-			msg = fmt.Sprintf("egress firewall in namespace %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		case libovsdbops.UDNIsolationOwnerType:
-			msg = fmt.Sprintf("UDN isolation of type %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
+func newACLEvent(o *nbdb.ACL) *model.ACLEvent {
+	actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
+	event := model.ACLEvent{
+		Action: o.Action,
+		Actor:  actor,
+	}
+	switch actor {
+	case libovsdbops.NetworkPolicyOwnerType:
+		nsname := strings.SplitN(o.ExternalIDs[libovsdbops.ObjectNameKey.String()], ":", 2)
+		if len(nsname) == 2 {
+			event.Namespace = nsname[0]
+			event.Name = nsname[1]
+		} else {
+			// do not split (which use cases?)
```

this should never happen, you can return an error is it does

---

done in https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4818/commits/b8ab657385ff1c6c21c80ed998cbb6cdaa59af3b